### PR TITLE
Fixed issue with the catalina_wrapper.sh

### LINKED ISSUE #46: catalina_wrapper.sh is missing in the Dockerfile.
Running the Geoserver docker image with the variable `GEOSERVER_APP_NAME="backend"`  and checking that the folder inside the container, I noticed that the directory remains with the default name `GeoServer`, so the URL doesn't work.
investigating the cause, I saw that the `catalina_wrapper.sh` is missing in the Dockerfile.

### PR DESCRIPTION
**Pull request by agpenton**
Resolves #46 
- catalina_wrapper.sh added to Dockerfile.
- added creation of plugins folder in custom_build.sh
- added .dockerignore file

### CODE REVIEW

File: custom_build.sh [hunk 1]
```diff
@@ -175,10 +177,13 @@ function main {
...
     download_geoserver "${GEOSERVER_VERSION}"
     clean_up_directory ${PLUGIN_ARTIFACT_DIRECTORY}
+    create_plugins_folder
+    download_plugin ext feature-pregeneralized
+    download_plugin ext css
     download_plugin ext monitor
     download_plugin ext control-flow
-    download_plugin community sec-oauth2-geonode
-    #download_marlin
+    download_plugin ext libjpeg-turbo
```
Comment (randomorder) [inline custom_build.sh:185]: please remove this

File: custom_build.sh [hunk 2]
```diff
@@ -175,10 +177,13 @@ function main {
     help ${ALL_PARAMETERS}
     download_geoserver "${GEOSERVER_VERSION}"
     clean_up_directory ${PLUGIN_ARTIFACT_DIRECTORY}
+    create_plugins_folder
+    download_plugin ext feature-pregeneralized
```
Comment (randomorder) [inline custom_build.sh:181]: not on master. Use a project specific branch instead

File: custom_build.sh [hunk 3]
```diff
@@ -175,10 +177,13 @@ function main {
...
+    download_plugin ext css
```
Comment (randomorder) [inline custom_build.sh:182]: not on master. Use a project specific branch instead

File: custom_build.sh [hunk 4]
```diff
@@ -147,8 +151,7 @@ function build_with_data_dir() {
         else
                 DOCKER_BUILD_COMMAND="docker build"
         fi;
-	${DOCKER_BUILD_COMMAND} --no-cache \
-		--build-arg GEOSERVER_WEBAPP_SRC=${GEOSERVER_ARTIFACT_DIRECTORY}/geoserver.war \
+	${DOCKER_BUILD_COMMAND} --build-arg GEOSERVER_WEBAPP_SRC=${GEOSERVER_ARTIFACT_DIRECTORY}/geoserver.war \
```
Comment (randomorder) [inline custom_build.sh:154]: why this change?

File: custom_build.sh [hunk 5]
```diff
@@ -164,8 +167,7 @@ function build_without_data_dir() {
 	else
 		DOCKER_BUILD_COMMAND="docker build"
 	fi;
-	${DOCKER_BUILD_COMMAND} --no-cache \
-		--build-arg GEOSERVER_WEBAPP_SRC=${GEOSERVER_ARTIFACT_DIRECTORY}/geoserver.war \
+	${DOCKER_BUILD_COMMAND} --build-arg GEOSERVER_WEBAPP_SRC=${GEOSERVER_ARTIFACT_DIRECTORY}/geoserver.war \
```
Comment (randomorder) [inline custom_build.sh:170]: why this change?

File: Dockerfile [hunk 6]
```diff
@@ -102,9 +102,10 @@ RUN apt-get update \
 
 COPY --from=mother "/opt/libjpeg-turbo" "/opt/libjpeg-turbo"
 COPY --from=mother "/output/datadir" "${GEOSERVER_DATA_DIR}"
-COPY --from=mother "/output/webapp/geoserver" "${CATALINA_BASE}/webapps/geoserver"
+#COPY --from=mother "/output/webapp/geoserver" "${CATALINA_BASE}/webapps/geoserver"
```
Comment (randomorder) [inline Dockerfile:105]: why this change?

**Review by randomorder — CHANGES_REQUESTED** [review]
- Some changes are ok
- Some are project specific and shouldn't end up on master. Please use a dedicated branch with the project code
- Some changes are not directly related with catalina_wrapper and is not clear why you're making them. Plese create separate issues and separate PRs as needed

File: custom_build.sh (hunk 1 above)
Comment (agpenton) [reply custom_build.sh:185]: is the branch ready to create the PR?

**Review by agpenton — COMMENTED** [review]
there are the answers to the changes.

**Pull request closed without merging**